src/common/runtime/helper/test_worker.ts

```diff
@@ -10,22 +10,27 @@ import { WorkerTestRunRequest } from './utils_worker.js';
 
 /** Query all currently-registered service workers, and unregister them. */
 function unregisterAllServiceWorkers() {
-  void navigator.serviceWorker.getRegistrations().then(registrations => {
-    for (const registration of registrations) {
-      void registration.unregister();
-    }
-  });
+  if ('serviceWorker' in navigator) {
+    void navigator.serviceWorker.getRegistrations().then(registrations => {
+      for (const registration of registrations) {
+        void registration.unregister();
+      }
+    });
+  }
 }
 
-// Firefox has serviceWorkers disabled in private mode
-// and Servo does not support serviceWorkers yet.
-if ('serviceWorker' in navigator) {
-  // NOTE: This code runs on startup for any runtime with worker support. Here, we use that chance to
-  // delete any leaked service workers, and register to clean up after ourselves at shutdown.
-  unregisterAllServiceWorkers();
-  window.addEventListener('beforeunload', () => {
-    unregisterAllServiceWorkers();
-  });
+// NOTE: This code runs on startup for any runtime with worker support. Here, we use that chance
+// to delete any leaked service workers (in case they weren't cleaned up by the 'beforeonload'
+// handler below).
+unregisterAllServiceWorkers();
+
+/**
+ * Add a 'beforeunload' handler. (We use this here instead of registerShutdownTask so that we don't
```

Comment:
I wonder if we should use `pagehide` instead as `beforeunload` has issues: https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event
See https://developer.chrome.com/docs/web-platform/page-lifecycle-api#the_beforeunload_event as well

Reply:
It sounds like the main reason not to use beforeunload is because it disables bfcache, but we don't care about that.

But maybe we should register both, if it trigger more reliably?